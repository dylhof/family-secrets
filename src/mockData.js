export const mockUser = {
  id: 1,
  name: 'Betty',
}


export const families = [
  {
    name: 'Reitmeyer',
    id: 2,
    user_id: 1
  },
  {
    name: 'Hofmann',
    id: 3,
    user_id: 1
  }
]

