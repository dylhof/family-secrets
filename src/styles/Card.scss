.Card--div {
  height: 300px;
  width: 200px;
  border: 2px solid $dark-accent-color;
  border-radius: 5px;

  img {
    width: 200px;
    max-height: 250px;
    overflow: scroll;
  }

  .Card--p--story{
    height: 250px;
    overflow: scroll;
    font-family: $body-font;
    font-size: 13px;
    padding: 5px;
  }

  h3 {
    font-family: $title-font;
    font-size: 20px;
    color: $main-color;
    text-align: center;
    margin: 5px;
  }
}