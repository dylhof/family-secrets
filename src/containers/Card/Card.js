import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as mockData from '../../mockData';

export class Card extends Component {
  generateCard = () => {
    const { currentView, title, story, name, ingredients, instructions, photo, caption} = this.props
    switch(currentView) {
      case 'Stories':
        return(
          <div className='Card--div'>
            <h3>{title}</h3>
            <p className='Card--p--story'>{story}</p>
          </div>
        ) 
      case 'Recipes':
        return(
          <div className='Card--div'>
            <h3>{name}</h3>
            <p>ingredients</p>
            <div>{this.generateIngredients(ingredients)}</div>
            <p>instructions</p>
            <p>{instructions}</p>
          </div>
        )
      case 'Photos':
        return (
          <div className='Card--div'>
            <img src={photo} alt={caption}/>
            <p>{caption}</p>
          </div>
        )
      default: 
        return ''
    }
  }

  generateIngredients = (ingredients) => {
    const splitIngredients = ingredients.split('\n');
    const ingredientsToShow = splitIngredients.map(ingredient => {
      return <li>{ingredient}</li>
    })
    return(
      <ul>
        {ingredientsToShow}
      </ul>
    )
  }

  render() {
    return(
      <div>
        {this.generateCard()}
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  currentView: state.currentView
});

export default connect(mapStateToProps)(Card);