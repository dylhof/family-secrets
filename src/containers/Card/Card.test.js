import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';
import * as mockData from '../../mockData';

describe('Card', () => {
  it('should match the snapshot if the current View is Stories', () => {
    const mockStory = mockData.stories[1];
    const mockCurrentView = 'Stories';
    const wrapper = shallow(
      <Card 
        currentView={mockCurrentView}
        title={mockStory.title}
        story={mockStory.story}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if the current View is Recipes', () => {
    const mockRecipe = mockData.recipes[1];
    const mockCurrentView = 'Recipes';
    const wrapper = shallow(
      <Card 
        currentView={mockCurrentView}
        name={mockRecipe.name}
        ingredients={mockRecipe.ingredients}
        directions={mockRecipe.directions}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if the current View is Stories', () => {
    const mockPhoto = mockData.photos[1];
    const mockCurrentView = 'Photos';
    const wrapper = shallow(
      <Card 
        currentView={mockCurrentView}
        photo={mockPhoto.photo}
        caption={mockPhoto.caption}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

});