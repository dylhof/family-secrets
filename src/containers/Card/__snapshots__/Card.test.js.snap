// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if the current View is Recipes 1`] = `
<div>
  <div
    className="Card--div"
  >
    <h3>
      soup
    </h3>
    <p>
      carrots broth chicken salt pepper noodles
    </p>
    <p>
      do some stuff Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc leo nunc, dignissim et bibendum pulvinar, tristique in nisi. Phasellus consectetur ullamcorper libero, eu scelerisque nunc finibus ut. Pellentesque in purus laoreet, tristique leo sed, sodales nisl. Nullam aliquet et nunc sit amet cursus. Phasellus tempor mauris ac orci tincidunt, a lacinia massa iaculis. Phasellus ipsum nisi, bibendum ac maximus hendrerit, congue quis felis. Curabitur quam nisl, tempor vel dolor a, suscipit tincidunt nisl. Phasellus leo ex, dictum id dapibus ut, egestas sed massa. Integer et lacinia felis. Nullam dui risus, suscipit in finibus nec, lacinia nec erat. Curabitur non justo purus. Nam at tempor arcu. Aliquam pharetra at nisi in placerat. Sed dignissim fermentum dui, ut feugiat nunc rutrum ac.
    </p>
  </div>
</div>
`;

exports[`Card should match the snapshot if the current View is Stories 1`] = `
<div>
  <div
    className="Card--div"
  >
    <h3>
      the family pony
    </h3>
    <p
      className="Card--p--story"
    >
      do some stuff Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc leo nunc, dignissim et bibendum pulvinar, tristique in nisi. Phasellus consectetur ullamcorper libero, eu scelerisque nunc finibus ut. Pellentesque in purus laoreet, tristique leo sed, sodales nisl. Nullam aliquet et nunc sit amet cursus. Phasellus tempor mauris ac orci tincidunt, a lacinia massa iaculis. Phasellus ipsum nisi, bibendum ac maximus hendrerit, congue quis felis. Curabitur quam nisl, tempor vel dolor a, suscipit tincidunt nisl. Phasellus leo ex, dictum id dapibus ut, egestas sed massa. Integer et lacinia felis. Nullam dui risus, suscipit in finibus nec, lacinia nec erat. Curabitur non justo purus. Nam at tempor arcu. Aliquam pharetra at nisi in placerat. Sed dignissim fermentum dui, ut feugiat nunc rutrum ac.
    </p>
  </div>
</div>
`;

exports[`Card should match the snapshot if the current View is Stories 2`] = `
<div>
  <div
    className="Card--div"
  >
    <img
      alt="wall cat"
      src="https://i.imgur.com/dRxnay8.jpg"
    />
    <p>
      wall cat
    </p>
  </div>
</div>
`;
